bzt/modules/selenium.py

```diff
@@ -1218,57 +1222,69 @@ def build_source_code(self):
+        test_class.append(self.gen_statement("driver=None", indent=4))
+        test_class.append(self.gen_new_line())
         test_class.append(self.gen_setupclass_method())
         test_class.append(self.gen_teardownclass_method())
+        test_class.append(self.gen_setup_method())
 
         counter = 0
         methods = {}
-        requests = self.scenario.get_requests()
-        scenario_timeout = self.scenario.get("timeout", 30)
+        requests = self.scenario.get_requests(False)
         default_address = self.scenario.get("default-address", None)
 
         for req in requests:
             if req.label:
                 label = req.label
-            else:
+            elif req.url:
                 label = req.url
+            else:
+                raise TaurusConfigError("You must specify at least 'url' or 'label' for each requests item")
             mod_label = re.sub('[^0-9a-zA-Z]+', '_', label[:30])
             method_name = 'test_%05d_%s' % (counter, mod_label)
             test_method = self.gen_test_method(method_name)
             methods[method_name] = label
             counter += 1
             test_class.append(test_method)
 
-            parsed_url = parse.urlparse(req.url)
-            if default_address is not None and not parsed_url.netloc:
-                url = default_address + req.url
-            else:
-                url = req.url
+            if req.url is not None:
+                self._add_url_request(default_address, req, test_method)
 
-            test_method.append(self.gen_comment("start request: %s" % url))
+            for action_config in req.config.get("actions", []):
+                test_method.append(self.gen_action(action_config))
 
-            if req.timeout is not None:
-                test_method.append(self.gen_impl_wait(req.timeout))
+            if "assert" in req.config:
+                test_method.append(self.gen_statement("body = self.driver.page_source"))
+                for assert_config in req.config.get("assert"):
+                    for elm in self.gen_assertion(assert_config):
+                        test_method.append(elm)
 
-            test_method.append(self.gen_statement("self.driver.get('%s')" % url))
             think_time = req.think_time if req.think_time else self.scenario.get("think-time", None)
-
             if think_time is not None:
                 test_method.append(self.gen_statement("sleep(%s)" % dehumanize_time(think_time)))
 
-            if "assert" in req.config:
-                test_method.append(self.__gen_assert_page())
-                for assert_config in req.config.get("assert"):
-                    test_method.extend(self.gen_assertion(assert_config))
-
-            if req.timeout is not None:
-                test_method.append(self.gen_impl_wait(scenario_timeout))
-
-            test_method.append(self.gen_comment("end request: %s" % url))
+            test_method.append(self.gen_statement("pass"))  # just to stub empty case
```

Comment:
Should we allow pretty such `requests` items?

I mean, if the item only has a `label` and doesn't have neither `url` nor `actions` — it's kind of meaningless.